packages/core/src/components/discord-reaction/discord-reaction.tsx

```diff
@@ -0,0 +1,50 @@
+import { Component, ComponentInterface, Element, h, Prop } from '@stencil/core';
+import clsx from 'clsx';
+
+@Component({
+	tag: 'discord-reaction',
+	styleUrl: 'discord-reaction.css'
+})
+export class DiscordReaction implements ComponentInterface {
+	/**
+	 * The DiscordReaction element.
+	 */
+	@Element()
+	public el: HTMLElement;
+
+	/**
+	 * The reaction emoji image URL.
+	 */
+	@Prop()
+	public emoji: string;
+
+	/**
+	 * The name of the emoji to use as alternative image text.
+	 */
+	@Prop()
+	public name = ':emoji:';
+
+	/**
+	 * The number of people who reacted.
+	 * @default 1
+	 */
+	@Prop()
+	public count = 1;
+
+	/**
+	 * Whether the reaction should show as reacted by the user.
+	 */
+	@Prop()
+	public reacted = false;
+
+	public render() {
+		return (
+			<div class={clsx('discord-reaction', { 'discord-reaction-reacted': this.reacted })}>
+				<div class="discord-reaction-inner">
+					<img src={this.emoji} alt={this.name} draggable={false} />
```

Comment:
I think it would be wise to scale down any provided src by providing `height` and `width`. To ensure that people do not put enormous images as reactions and get unexpected behaviour. You can test this by providing such a large image as the src (i.e. in `png` format) and checking the result.

Reply:
This is already set by the CSS styling:

```css
.discord-reaction img {
	width: 1rem;
	height: 1rem;
	margin: 0.125rem 0;
	min-width: auto;
	min-height: auto;
	object-fit: contain;
	vertical-align: bottom;
}
```

Setting the emoji as a big non-square picture still works as intended:
![](https://get.snaz.in/42QXuzS.png)

Reply:
ah ok, I missed that!